Sources/CognitoIdentityDemo/CognitoIdentityDemo.swift

```diff
@@ -47,66 +54,69 @@ class CognitoIdentityDemo {
+                        
+                        if let foundID = foundID {
+                            poolID = foundID
                         }
                     }
                     
                     token = output.nextToken
                 case .failure(let error):
-                    completion(nil)
                     print("ERROR: ", dump(error, name: "Error scanning identity pools"))
+                    token = nil
+                    poolID = nil
                 }
             }
-        } while token != nil
+        } while token != nil && poolID == nil
         
-        completion(nil)
+        completion(poolID)
     }
     
     /// Returns the ID of the identity pool with the specified name.
     /// - Parameters:
     ///   - name: The name of the identity pool whose ID should be returned
     /// - Returns: A string containing the ID of the specified identity pool or `nil` on error or if not found
-    func getIdentityPoolID(name: String,
+    func getIdentityPoolID(name: String, createIfMissing: Bool = false,
                            completion: @escaping (String?) -> Void) {
         // See if the pool already exists
         
-        self.getIdentityPoolID(name: name) { poolId in
-            if let poolId = poolId {
-                completion(poolId)
+        self.getIdentityPoolID(name: name) { foundID in
+            if let foundID = foundID {
+                completion(foundID)
+            } else if createIfMissing {
+                self.createIdentityPool(name: name, completion: completion)
             } else {
-                self.createIdentityPool(name: name) { poolId in
-                    completion(poolId)
-                }
+                completion(nil)
             }
         }
-    }
+}
     
     /// Create a new identity pool, returning its ID.
     /// - Parameters:
     ///     - name: The name to give the new identity pool
     /// - Returns: A string containing the newly created pool's ID, or `nil` if an error occurred
     func createIdentityPool(name: String, completion: @escaping (String?) -> Void) {
-        
         let cognitoInputCall = CreateIdentityPoolInput(
             developerProviderName: "com.exampleco.CognitoIdentityDemo",
             identityPoolName: name
         )
         
-        cognitoIdentityClient.createIdentityPool(input: cognitoInputCall) { (result) in
+        cognitoIdentityClient.createIdentityPool(input: cognitoInputCall) { result in
             switch(result) {
             case .success(let output):
-                if let poolId = output.identityPoolId {
-                    completion(poolId)
+                if let poolID = output.identityPoolId {
+                    completion(poolID)
                 }
-                completion(nil)
             case .failure(let error):
-                completion(nil)
                 print("ERROR: ", dump(error, name: "Error attempting to create the identity pool"))
```

Comment:
Is there a strong reason to use `print` over swift-log?  we can easily do something like:
```
import Logging

...
    let logger: Logger = Logger(label: "CognitoIdentityDemo")
```
and then:
```
logger.error("....")
```

Reply:
The main reason I'm using print is that I figured doing so separates my output from the logging, since my code is supposed to be a stand-in for some kind of functional code rather than a true log entry.